src/Components/test/E2ETestMigration/Tests/InputFileTest.cs

```diff
@@ -0,0 +1,260 @@
+            }
+
+            var file = TempFile.Create(_tempDirectory, "txt", contentBuilder.ToString());
+
+            await VerifyFile(file);
+        }
+
+        [Fact]
+        public async Task CanUploadMultipleFiles()
+        {
+            // Create multiple small text files
+            var files = Enumerable.Range(1, 3)
+                .Select(i => TempFile.Create(_tempDirectory, "txt", $"Contents of file {i}."))
+                .ToList();
+
+            await VerifyFiles(files);
+        }
+
+        [Fact]
+        public async Task CanUploadAndConvertImageFile()
+        {
+            var sourceImageId = "image-source";
+
+            // Get the source image base64
+            var base64 = await _page.EvaluateAsync<string>($@"
+                const canvas = document.createElement('canvas');
+                const context = canvas.getContext('2d');
+                const image = document.getElementById('{sourceImageId}');
+
+                canvas.width = image.naturalWidth;
+                canvas.height = image.naturalHeight;
+                context.drawImage(image, 0, 0, image.naturalWidth, image.naturalHeight);
+
+                canvas.toDataURL().split(',').pop();");
+
+            // Save the image file locally
+            var file = TempFile.Create(_tempDirectory, "png", Convert.FromBase64String(base64));
+
+            // Re-upload the image file (it will be converted to a JPEG and scaled to fix 640x480)
+            var inputFile = await _page.QuerySelectorAsync("#input-image");
+            await inputFile.SetInputFilesAsync(file.Path);
+
+            // Validate that the image was converted without error and is the correct size
+            var uploadedImage = await _page.WaitForSelectorAsync("#image-uploaded");
+            Assert.NotNull(uploadedImage);
+            var box = await uploadedImage.GetBoundingBoxAsync();
+            Assert.Equal(480, box.Height);
+            Assert.Equal(480, box.Width);
+        }
+
+        protected async Task ClearAndType(string selector, string value)
+        {
+            await _page.EvalOnSelectorAsync(selector, "e => e.Value = ''");
+            var element = await _page.QuerySelectorAsync(selector);
+            await element.TypeAsync(value);
+        }
+
+        [Fact]
+        public async Task ThrowsWhenTooManyFilesAreSelected()
+        {
+            await ClearAndType("#max-allowed-files", "1\n");
+
+            // Save two files locally
+            var file1 = TempFile.Create(_tempDirectory, "txt", "This is file 1.");
+            var file2 = TempFile.Create(_tempDirectory, "txt", "This is file 2.");
+
+            // Select both files
+            await _page.SetInputFilesAsync("#input-file", new string[] { file1.Path, file2.Path });
+
+            // Validate that the proper exception is thrown
+            var exceptionMessage = await _page.QuerySelectorAsync("#exception-message");
```

Comment:
review: exception message selector are not showing up, do these need to wait/for selector?

Reply:
e.Value != e.value....grr casing with no errors